cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java

```diff
@@ -109,9 +111,16 @@
 
   @Nullable
   @Override
-  protected RuntimeInfo createRuntimeInfo(ProgramController controller, ProgramId programId) {
+  protected RuntimeInfo createRuntimeInfo(ProgramController controller, ProgramId programId,
+                                          final Runnable cleanUpTask) {
     if (controller instanceof AbstractTwillProgramController) {
       RunId twillRunId = ((AbstractTwillProgramController) controller).getTwillRunId();
+      controller.addListener(new AbstractListener() {
+        @Override
+        public void alive() {
```

Comment:
You might want to capture the `init` method as well as the state can already be `ALIVE`.

Reply:
Added for `init` as well.